supabase/functions/mailgun/index.ts

```diff
@@ -0,0 +1,184 @@
+    if (!sales)
+        return new Response(
+            `Unable to find sales in database, email: ${salesEmail}`,
+            { status: 406 }
+        );
+
+    const { firstName, lastName, email, domain } = extractMailContactData(
+        to.toString()
+    );
+    if (!email)
+        return new Response(`Could not extract email from To: ${to}`, {
+            status: 406,
+        });
+
+    const { data: existingContact, error: fetchContactError } =
+        await supabaseAdmin
+            .from('contacts')
+            .select('*')
+            .eq('email', email)
+            .maybeSingle();
+    if (fetchContactError)
+        return new Response(
+            `Could not fetch contact from database, email: ${email}`,
+            { status: 500 }
+        );
+
+    // deno-lint-ignore no-explicit-any
+    let contact: any = undefined;
+    if (existingContact) {
+        contact = existingContact;
+    } else {
+        // If the contact does not exist, we need to create it, along with the company if needed
+        const { data: existingCompany, error: fetchCompanyError } =
+            await supabaseAdmin
+                .from('companies')
+                .select('*')
+                .eq('name', domain)
+                .maybeSingle();
+        if (fetchCompanyError)
+            return new Response(
+                `Could not fetch companies from database, name: ${domain}`,
+                { status: 500 }
+            );
+
+        // deno-lint-ignore no-explicit-any
+        let company: any = undefined;
+        if (existingCompany) {
+            company = existingCompany;
+        } else {
+            const { data: newCompanies, error: createCompanyError } =
+                await supabaseAdmin
+                    .from('companies')
+                    .insert({ name: domain, sales_id: sales.id })
+                    .select();
+            if (createCompanyError)
+                return new Response(
+                    `Could not create company in database, name: ${domain}`,
+                    { status: 500 }
+                );
+            company = newCompanies[0];
+        }
+
+        // Create the contact
+        const { data: newContacts, error: createContactError } =
+            await supabaseAdmin
+                .from('contacts')
+                .insert({
+                    first_name: firstName,
+                    last_name: lastName,
+                    email,
+                    company_id: company.id,
+                    sales_id: sales.id,
+                    first_seen: new Date(),
+                    last_seen: new Date(),
+                    tags: [],
+                })
+                .select();
+        if (createContactError || !newContacts[0])
+            return new Response(
+                `Could not create contact in database, email: ${email}`,
+                { status: 500 }
+            );
+        contact = newContacts[0];
+    }
+
+    // Add note to new contact
```

Comment:
You don't know if this is a new contact at this stage - the comment is misleading

Reply:
probably a leftover from a prior refactor to the code, nice catch!